modules/webhook/matrix.go

```diff
@@ -0,0 +1,285 @@
+		if err != nil {
+			return nil, err
+		}
+
+		text = fmt.Sprintf("[%s] Pull request review %s: [%s](%s) by %s", repoLink, action, title, titleLink, senderLink)
+	}
+
+	return &MatrixPayload{
+		Body:          getMessageBody(text),
+		MsgType:       messageTypeText[matrix.MessageType],
+		Format:        "org.matrix.custom.html",
+		FormattedBody: text,
+	}, nil
+}
+
+func getMatrixRepositoryPayload(p *api.RepositoryPayload, matrix *MatrixMeta) (*MatrixPayload, error) {
+	senderLink := MatrixLinkFormatter(setting.AppURL+p.Sender.UserName, p.Sender.UserName)
+	repoLink := MatrixLinkFormatter(p.Repository.HTMLURL, p.Repository.FullName)
+	var text string
+
+	switch p.Action {
+	case api.HookRepoCreated:
+		text = fmt.Sprintf("[%s] Repository created by %s", repoLink, senderLink)
+	case api.HookRepoDeleted:
+		text = fmt.Sprintf("[%s] Repository deleted by %s", repoLink, senderLink)
+	}
+
+	return &MatrixPayload{
+		Body:          getMessageBody(text),
+		MsgType:       messageTypeText[matrix.MessageType],
+		Format:        "org.matrix.custom.html",
+		FormattedBody: text,
+	}, nil
+}
+
+// GetMatrixPayload converts a Matrix webhook into a MatrixPayload
+func GetMatrixPayload(p api.Payloader, event models.HookEventType, meta string) (*MatrixPayload, error) {
+	s := new(MatrixPayload)
+
+	matrix := &MatrixMeta{}
+	if err := json.Unmarshal([]byte(meta), &matrix); err != nil {
+		return s, errors.New("GetMatrixPayload meta json:" + err.Error())
+	}
+
+	switch event {
+	case models.HookEventCreate:
+		return getMatrixCreatePayload(p.(*api.CreatePayload), matrix)
+	case models.HookEventDelete:
+		return getMatrixDeletePayload(p.(*api.DeletePayload), matrix)
+	case models.HookEventFork:
+		return getMatrixForkPayload(p.(*api.ForkPayload), matrix)
+	case models.HookEventIssues, models.HookEventIssueAssign, models.HookEventIssueLabel, models.HookEventIssueMilestone:
+		return getMatrixIssuesPayload(p.(*api.IssuePayload), matrix)
+	case models.HookEventIssueComment, models.HookEventPullRequestComment:
+		return getMatrixIssueCommentPayload(p.(*api.IssueCommentPayload), matrix)
+	case models.HookEventPush:
+		return getMatrixPushPayload(p.(*api.PushPayload), matrix)
+	case models.HookEventPullRequest, models.HookEventPullRequestAssign, models.HookEventPullRequestLabel,
+		models.HookEventPullRequestMilestone, models.HookEventPullRequestSync:
+		return getMatrixPullRequestPayload(p.(*api.PullRequestPayload), matrix)
+	case models.HookEventPullRequestReviewRejected, models.HookEventPullRequestReviewApproved, models.HookEventPullRequestReviewComment:
+		return getMatrixPullRequestApprovalPayload(p.(*api.PullRequestPayload), matrix, event)
+	case models.HookEventRepository:
+		return getMatrixRepositoryPayload(p.(*api.RepositoryPayload), matrix)
+	case models.HookEventRelease:
+		return getMatrixReleasePayload(p.(*api.ReleasePayload), matrix)
+	}
+
+	return s, nil
+}
+
+var urlRegex = regexp.MustCompile(`<a.*?href="(.*?)">(.*?)</a>`)
```

Comment:
How about something a little stricter?

```suggestion
var urlRegex = regexp.MustCompile(`<a [^>]*?href="([^">]*?)">(.*?)</a>`)
```

This would prevent matching with:
```
<attr [....] <a href [...]
<a href="xxx> [.....] "> (invalid code, but...)
<a name="xxx"> [...] </a> <a href="xxx" [...]
```